### DISCUSSION TITLE
Don't allow deleting the root channel!

### DISCUSSION THREAD

**Discussion by slannon97** [BearWare/TeamTalk5#2499] [Ideas]
### Description

It's possible to delete the root channel, which deletes all files and channels under it. Please add a check so the server won't allow the deletion of /.

### Application

- [X] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkClassic
- [X] TeamTalkServer

### Platform

- [X] Windows
- [ ] macOS
- [ ] Android
- [ ] iOS
- [ ] Linux

### Expected behavior

The server should produce an error, or better yet, prevent the client from even allowing this.

### Actual behavior

The root channel and everything under it is immediately deleted.

### Steps to reproduce problem

Press F8 when focused on the top level channel or use the delete channel option in the channel menu. Respond yes to the delete prompt.

**Comment by amirmahdifard** [comment]
this is not a bug, it's a feature. This is needed to clear the server completely, raather than deleting everything one by one when your server has a lot of channels and subchannels and files in them. Actually the root channel will not be deleted and you can stil use the server, but everything will be cleared, this is for those times you want to transfer that server to someone else, or starting fresh for any reason
